src/security_buffer.rs

```diff
@@ -110,29 +245,64 @@ impl<'data> SecurityBuffer<'data> {
     /// If a slice contains more than one buffer with a specified buffer type, then the first one will be returned.
     pub fn find_buffer<'a>(
         buffers: &'a [SecurityBuffer<'data>],
-        buffer_type: SecurityBufferType,
+        buffer_type: BufferType,
     ) -> Result<&'a SecurityBuffer<'data>> {
+        buffers.iter().find(|b| b.buffer_type() == buffer_type).ok_or_else(|| {
+            Error::new(
+                ErrorKind::InvalidToken,
+                format!("no buffer was provided with type {:?}", buffer_type),
+            )
+        })
+    }
+
+    /// Returns the vector of immutable references to the [SecurityBuffer] with specified buffer type.
+    pub fn buffers_with_type<'a>(
```

Comment:
naming: `of` instead of `with` is clearer. Same suggestion for the other filtered iterators.
```suggestion
    pub fn buffers_of_type<'a>(
```

Reply:
done